src/theory/quantifiers/sygus/synth_conjecture.cpp

```diff
@@ -580,8 +593,30 @@ bool SynthConjecture::doCheck(std::vector<Node>& lems)
 
   if (!query.isConst() || query.getConst<bool>())
   {
+    // add recursive function definitions
+    FunDefEvaluator* feval = d_tds->getFunDefEvaluator();
+    const std::vector<Node>& fdefs = feval->getDefinitions();
+    if (!fdefs.empty())
+    {
+      // get the relevant definitions based on the symbols in the query
```

Comment:
How important is this? Like, if we were to pass formulas that ranged over symbols not in the query it'd be a non-issue? I ask because maybe this could be simplified then for us to always pass the recfuns relevant to all symbols in the grammar.

Reply:
This is important. You only want to pass the recursive functions you need. In the regression added here, this actually may make the difference of decidability vs undecidability in the verfication step: candidates that don't involve the recursive function have a decidable verification step, and CEGIS gets a proper counterexample point.

Reply:
I see. I had not thought about making a difference that way, it'd be good to add a comment clarifying.